src/command_modules/azure-cli-component/azure/cli/command_modules/component/custom.py

```diff
@@ -16,6 +16,10 @@
 COMPONENT_PREFIX = 'azure-cli-'
 
 
+def _deprecate_warning():
+    logger.warning("The 'component' commands will be deprecated in the future.")
```

Comment:
The warning message is more for users of the CLI rather than the developer.

I'd like the output to be like this:
```
$ az component list
The 'component' commands will be deprecated in the future.
```

Using `warnings.warn`, I either got nothing printed or this. Neither of which I want.
```
$ az component list
/Users/debekoe/Repos/derekbekoe-azure-cli/src/command_modules/azure-cli-component/azure/cli/command_modules/component/custom.py:21: UserWarning: The 'component' commands will be deprecated in the future.
  warnings.warn("The 'component' commands will be deprecated in the future.")
```